### PR TITLE
remove last uses of parking-lot

### PR DESCRIPTION
**Pull request by Geal**
use a futures aware mutex instead to avoid blocking executors

### CODE REVIEW

File: crates/apollo-router/Cargo.toml [hunk 1]
```diff
@@ -69,6 +68,7 @@ httpmock = "0.6.2"
 insta = "1.8.0"
 maplit = "1.0.2"
 mockall = "0.10.2"
+parking_lot = "0.11.2"
```
Comment (cecton) [inline crates/apollo-router/Cargo.toml:71]: I suppose that one is for the counter in the integration test? Maybe we use the std's one and use unwrap to remove the burden of maintaining a dependency just for a test?

**Review by cecton — APPROVED** [review]
That's a no brainer :grin:

**Review by BrynCooke — APPROVED** [review]

File: crates/apollo-router/Cargo.toml (hunk 1 above)
Comment (Geal) [reply crates/apollo-router/Cargo.toml:71]: good point,it'll be easier that way

Comment (Geal) [reply crates/apollo-router/Cargo.toml]: d16b35d

**Pull request merged**